src/custom/state/swap/extension.ts

```diff
@@ -0,0 +1,159 @@
+import { CurrencyAmount, Trade, Currency } from '@uniswap/sdk'
+import { useTradeExactIn, useTradeExactOut } from 'hooks/Trades'
+import { tryParseAmount } from './hooks'
+import { basisPointsToPercent } from 'utils'
+import { FeeInformation } from 'custom/utils/operator'
+
+interface DetermineFee {
+  inputAmount: CurrencyAmount
+  feeInformation: FeeInformation
+}
+
+/**
+ * DetermineFee
+ * @description using the inputAmount, compares whether the minimalFee or the feeRatio is greater and returns
+ * @description accepts fractional values
+ */
+function determineFee({ feeInformation, inputAmount }: DetermineFee): CurrencyAmount | undefined {
+  const { feeRatio, minimalFee } = feeInformation
+
+  // We need to consider some cases here:
+  // Case 1: No feeInformation - return null
+  // Case 2: No feeRatio, but minimalFee
+  //   Case2a: is minimalFee >= inputAmount? - return null
+  //   Case2b: minimalFee < inputAmount - return fee
+  // Case 3: feeRatio && minimalFee
+  //  Case 3a: feeRatio * inputAmount > minimalFee - return feeRatio
+  //  Case 3b: feeRatio * inputAmount <= minimalFee - return minimalFee
+  //  Case 3c: feeRatio exists && minimalFee > inputAmount - return feeRatio
+
+  // MINIMAL FEE
+  const minimalFeeAsCurrency = tryParseAmount(minimalFee, inputAmount.currency)
```

Comment:
Why aren't you useing the `utils/fee.ts` `getFeeAmount` which has already some unit tests and should give you the actual fee.

Reply:
yup, changed